net/dns/resolver/forwarder.go

```diff
@@ -59,9 +60,12 @@ func truncatedFlagSet(pkt []byte) bool {
 
 const (
 	// dohTransportTimeout is how long to keep idle HTTP
-	// connections open to DNS-over-HTTPs servers. This is pretty
-	// arbitrary.
-	dohTransportTimeout = 30 * time.Second
+	// connections open to DNS-over-HTTPs servers. 10 seconds is a
+	// sensible default, as it's long enough to handle a burst of
+	// queries coming in a row, but short enough to not keep idle
+	// connections open for too long (which affects battery life on
+	// mobile devices).
```

Comment:
Doing more TLS handshakes (CPU + radio) more often might be more painful than an idle connection (which if truly idle doesn't cost anything).

Can we A/B test this and/or gather data to see what good idle numbers would empirically be? What the gaps are between DNS lookups?

Reply:
Updated the comment to better explain this.

> Doing more TLS handshakes (CPU + radio) more often might be more painful than an idle connection

My intuition here is that if the phone is forwarding a DNS query, it indicates an attempt to reach something on the internet. In that case, the modem would have woken up regardless. Performing a TLS handshake while the modem is already active will have minimal impact. It’s the wakeups in the background, rather than the activity while the modem is on, that drain the battery. The purpose of this PR is to specifically minimize the battery impact of keepalive packets sent by DoH servers on long-lived idle connections, which wake up the modem to be received and ACK'd.